File: helper-package-js/action.js
const { FACE } = require('./constants');
const { Position } = require('./position');

exports.Action = class Action {
    constructor(options) {
        this.type = options.type;
        this.moveOrSpawnAmount = options.move || options.amount || 0;
        this.face = options.face || FACE.UP;
        this.pos = options.pos || new Position();
    }
};

File: helper_package_python/action.py
from helper_package_python.position import Position
from helper_package_python.constants import Face

class Action:
	def __init__(self, type, pos=None, moveOrSpawnAmount=None, face=None):
		self.type = type
		self.pos = pos or Position()
		self.moveOrSpawnAmount = moveOrSpawnAmount or 0
		self.face = face or Face.UP
